var/spack/repos/builtin/packages/plumed/package.py

```diff
@@ -68,13 +69,61 @@ class Plumed(AutotoolsPackage):
     # Variants. PLUMED by default builds a number of optional modules.
     # The ones listed here are not built by default for various reasons,
     # such as stability, lack of testing, or lack of demand.
-    # FIXME: This needs to be an optional
+    #
+    # From 'configure --help' @2.3:
+    # all/none/reset or : separated list such as
+    # +crystallization:-bias default: reset
+    #
+    # Optional modules can be provided in two ways, via the `optional_modules` variant:
+    # 1. Use a reference set of optional modules via `optional_modules` (recommended).
+    #    Allowed values are: `all`[default], `reset`.
+    # 2. Pick any combination of specific optional modules (advanced).
+    #    Only the requested optional modules will be activated.
+    #    See list in variable `single_optional_modules` below.
+    #    This list comes from the Plumed manual, eg for 2.8:
+    #    https://www.plumed.org/doc-v2.8/user-doc/html/mymodules.html
+    # These are implemented using multi-valued variants (`disjoint_sets`),
+    # and the `conditional` option to handle version conflicts.
+    single_optional_modules = (conditional('adjmat', when='@2.3:'),
+                               'analysis',
+                               conditional('annfunc', when='@2.6:'),
+                               'bias', 'cltools', 'colvar', 'crystallization',
+                               conditional('dimred', when='@2.5:'),
+                               conditional('drr', when='@2.4:'),
+                               conditional('eds', when='@2.4:'),
+                               conditional('fisst', when='@2.7:'),
+                               'function',
+                               conditional('funnel', when='@2.7:'),
+                               'generic',
+                               conditional('imd', when='@:2.2.99'),
+                               conditional('isdb', when='@2.4:'),
+                               conditional('logmfd', when='@2.5:'),
+                               'manyrestraints', 'mapping',
+                               conditional('maze', when='@2.6:'),
+                               'molfile',
+                               'multicolvar',
+                               conditional('opes', when='@2.7:'),
+                               conditional('pamm',
+                               when='@2.5: optional_modules=adjmat'),
```

Comment:
That's nitpicking but I would align `when=...` with `'pamm'.

Also is the concretizer really clever enough to deal with this (I mean you have a multivalued variant with a value depending on another value of that same variant)? If it is then it's great!

Reply:
Oh I see!
Just done it now with a new commit.